### PR TITLE
Core: Add stack trace to ErrorResponse

### PR DESCRIPTION
**Pull request by rdblue**
#4367 added a way to send stack traces in REST error responses. This updates the `ErrorResponse` implementation and test code to send stack traces.

### CODE REVIEW

File: core/src/main/java/org/apache/iceberg/rest/responses/ErrorResponseParser.java [hunk 1]
```diff
@@ -64,6 +66,13 @@ public static void toJson(ErrorResponse errorResponse, JsonGenerator generator)
     generator.writeStringField(MESSAGE, errorResponse.message());
     generator.writeStringField(TYPE, errorResponse.type());
     generator.writeNumberField(CODE, errorResponse.code());
+    if (errorResponse.stack() != null) {
+      generator.writeArrayFieldStart(STACK);
+      for (String line : errorResponse.stack()) {
+        generator.writeString(line);
+      }
+      generator.writeEndArray();
```
Comment (kbendick) [inline core/src/main/java/org/apache/iceberg/rest/responses/ErrorResponseParser.java:74]: Nit: Should we consider writing the `stack` field even if it's `null`?

**Review by kbendick — COMMENTED** [review]
Overall this looks good to me.

I left a small nit about writing out the `stack` field to the JSON even if it is `null`, but I'm leaning more towards not adding it for now.

We should update the OpenAPI document though.

**Review by danielcweeks — APPROVED** [review]

File: core/src/main/java/org/apache/iceberg/rest/responses/ErrorResponseParser.java (hunk 1 above)
Comment (rdblue) [reply core/src/main/java/org/apache/iceberg/rest/responses/ErrorResponseParser.java:74]: I think it is good this way. Why send extra information that is already clear?

**Comment by rdblue** [comment]
Thanks for reviewing, @danielcweeks and @kbendick!

**Pull request merged**